File: dataset/human/c/file_2752.c
#include <stdio.h>

int main(void)
{
	int l,m,n,K,S,num=0;
	scanf("%d%d",&K,&S);

	for(l=0;l<=K;l++)
		for(m=l;m<=K;m++)
			for(n=m;n<=K;n++){
				if(l+m+n==S){
					if(l==m&&m==n){
						num++;
					}else if(l!=m&&l!=n&&m!=n){
						num+=6;
					}else{
						num+=3;
					}
				}
			}
	
	printf("%d\n",num);
	return 0;
}

File: dataset/human/python/file_2752.py
io = STDIN
d,G=io.gets.split.map(&:to_i)
ar=d.times.map do |i|
  q,c = io.gets.split.map(&:to_i)
  [100*(i+1),q,100*(i+1)*q+c] #点数、問題数、合計
end
$ans=1000
def dfs(questions,highest,amt,num)
  if amt>=G
    $ans=[$ans,num].min
  else
    if questions==[]
      x,q,tot=highest
      if G-amt<=x*(q-1)
        $ans=[$ans,num+((G-amt).to_f/x).ceil].min
      end
      return
    end
    x,q,tot=questions[0]
    dfs(questions[1..-1],highest,amt+tot,num+q)
    dfs(questions[1..-1],questions[0],amt,num)
  end
end
dfs(ar,[],0,0)
p $ans
